core/src/main/java/org/elasticsearch/script/ScriptMetaData.java

```diff
@@ -0,0 +1,292 @@
+import org.elasticsearch.cluster.DiffableUtils;
+import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.common.ParseFieldMatcher;
+import org.elasticsearch.common.ParsingException;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentParser.Token;
+import org.elasticsearch.common.xcontent.XContentType;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+public final class ScriptMetaData implements MetaData.Custom {
+
+    public final static String TYPE = "stored_scripts";
+    public final static ScriptMetaData PROTO = new ScriptMetaData(Collections.emptyMap());
+
+    private final Map<String, ScriptAsBytes> scripts;
+
+    ScriptMetaData(Map<String, ScriptAsBytes> scripts) {
+        this.scripts = scripts;
+    }
+
+    public BytesReference getScriptAsBytes(String language, String id) {
+        ScriptAsBytes scriptAsBytes = scripts.get(toKey(language, id));
+        if (scriptAsBytes != null) {
+            return scriptAsBytes.script;
+        } else {
+            return null;
+        }
+    }
+
+    public String getScript(String language, String id) {
+        BytesReference scriptAsBytes = getScriptAsBytes(language, id);
+        if (scriptAsBytes == null) {
+            return null;
+        }
+
+        try (XContentParser parser = XContentHelper.createParser(scriptAsBytes)) {
+            XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON);
+            parser.nextToken();
+            parser.nextToken();
+            switch (parser.currentName()) {
+                case "script":
+                    if (parser.nextToken() == Token.VALUE_STRING) {
+                        return parser.text();
+                    } else {
+                        builder.copyCurrentStructure(parser);
+                    }
+                    break;
+                case "template":
+                    if (parser.nextToken() == Token.VALUE_STRING) {
+                        return parser.text();
+                    } else {
+                        builder.copyCurrentStructure(parser);
+                    }
+                    break;
+                default:
+                    builder.startObject();
+                    builder.copyCurrentStructure(parser);
+                    break;
+            }
+            return builder.string();
```

Comment:
The reason I did this is because we used to do this before too (see master, `ScriptService#getScriptFromResponse(...)`). We don't and didn't store the encapsulating `script` or `template` objects but upon returning the script via the get scripts api we do include this.